x-pack/plugins/endpoint/public/applications/endpoint/index.tsx

```diff
@@ -21,8 +21,14 @@ import { HostList } from './view/hosts';
 import { PolicyList } from './view/policy';
 import { PolicyDetails } from './view/policy';
 import { HeaderNavigation } from './components/header_nav';
+import { setupIngestManager } from '../../../../ingest_manager/common';
 import { EuiThemeProvider } from '../../../../../legacy/common/eui_styled_components';
 
+export async function setupDeps(coreStart: CoreStart) {
+  // TODO handle a 500 from the ingest manager
+  await setupIngestManager(coreStart.http);
```

Comment:
@oatkiller @kqualters-elastic if the ingest manager fails to setup, what should we do? Maybe display a message to the user? This would be a critical failure.

Reply:
My suggestion would be to call and handle the setup here in this file rather than in `plugin.ts`.  We should show a loader while setup is being called - if that fails, then we can show an Error right on the page. Changes would likely go into the `AppRoot` component below - something like:

```typescript
const AppRoot: React.FunctionComponent<RouterProps> = React.memo(
  ({
    basename,
    store,
    coreStart: { http, notifications, uiSettings, application },
    depsStart: { data },
  }) => {
    const isDarkMode = useObservable<boolean>(uiSettings.get$('theme:darkMode'));

    const [isSetupDone, setIsSetupDone] = useState(false);
    const [wasSetupSuccessful, setWasSetupSuccessful] = useState(true);
    const [ingestSetupResponse, setIngestSetupResponse] = useState(undefined);

    useEffect(() => {
      setupIngestManager()
        .then(() => {
          setIsSetupDone(true);
        })
        .catch(error => {
          setWasSetupSuccessful(false);
          setIngestSetupResponse(error);
        });
    }, []);

    if (!isSetupDone) {
      return <EuiLoading />;
    }

    if (!wasSetupSuccessful) {
      return <EuiCallOut>show error here from {ingestSetupResponse}</EuiCallOut>;
    }

    // ...
```

I would also suggest storing a module level variable that tracks whether the setup was already called - so that you don't have to do it again when the user re-visits the App.

Reply:
@jonathan-buttner at what interval do we expect the retrieved index patterns to change?

Reply:
@oatkiller it won't be very often. It might need to change when we have a breaking change in a mapping. Do you think it'd be better to leave it hardcoded? We'd just have to maintain it in both places when it does change.